src/jwt.rs

```diff
@@ -82,6 +70,26 @@ pub struct Actor {
     pub provider: bool,
 }
 
+/// The claims metadata corresponding to a capability provider
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, Default)]
+pub struct CapabilityProvider {
+    /// A descriptive name for the capability provider
+    pub name: Option<String>,
+    /// The capability contract ID this provider supports
+    pub capid: String,
+    /// A human-readable string identifying the vendor of this provider (e.g. Redis or Cassandra or NATS etc)
+    pub vendor: String,
+    /// Indicates a monotonically increasing revision number.  Optional.
+    #[serde(rename = "rev", skip_serializing_if = "Option::is_none")]
+    pub rev: Option<i32>,
+    /// Indicates a human-friendly version string. Optional.
+    #[serde(rename = "ver", skip_serializing_if = "Option::is_none")]
+    pub ver: Option<String>,
+    /// The file hashes that correspond to the achitecture-OS target triples for this provider.
+    pub target_hashes: HashMap<String, String>
```

Comment:
Do we want to have this as a `HashMap<Architecture, String>`, where `Architecture` is an enum of supported architectures? I believe the idea here is that we won't have a strict set of architectures that we support, but if we only provide capability providers for a certain set then it would be valuable to prevent architecture typos

Reply:
I think in this particular crate we want it to just be strings. I don't want to have to deal with potential breaking changes to the claims format when we add or remove supported architectures and target operating systems. I think tooling that reads/writes these claims is probably what should be responsible for validating the target strings at the time.

Reply:
That makes sense to me